backend/tracim_backend/models/auth.py

```diff
@@ -87,14 +88,45 @@ class AuthType(enum.Enum):
     REMOTE = "remote"
 
 
+class Profile(enum.Enum):
+    """This model is the "max" group associated to a given user."""
+
+    NOBODY = (0, "nobody")
+    USER = (1, "users")
+    TRUSTED_USER = (2, "trusted-users")
+    ADMIN = (3, "administrators")
+
+    def __init__(self, profile_id: int, slug: str):
+        self.id = profile_id
+        self.slug = slug
+
+    @classmethod
+    def get_all_valid_slugs(self):
+        return [item.slug for item in list(Profile) if item.id > 0]
+
+    @classmethod
+    def get_profile_from_id(self, id: int):
+        profiles = [item for item in list(Profile) if item.id == id]
+        if len(profiles) != 1:
+            raise ProfileDoesNotExist()
+        return profiles[0]
+
+    @classmethod
+    def get_profile_from_slug(self, slug: str):
+        profiles = [item for item in list(Profile) if item.slug == slug]
+        if len(profiles) != 1:
+            raise ProfileDoesNotExist()
+        return profiles[0]
```

Comment:
For these 3 class methods:

* thirst parameter name must be `cls` and not `self` (PyCharm should underline it in yellow)
* You can replace `list(Profile)` by `list(cls)` for better maintainability

Reply:
done.